src/buffer/out/TextColor.h

```diff
@@ -82,7 +82,8 @@ struct TextColor
     static constexpr size_t FRAME_FOREGROUND = 263;
     static constexpr size_t FRAME_BACKGROUND = 264;
     static constexpr size_t CURSOR_COLOR = 265;
-    static constexpr size_t TABLE_SIZE = 266;
+    static constexpr size_t SELECTION_BACKGROUND = 266;
+    static constexpr size_t TABLE_SIZE = 267;

```

Comment:
If there's a chance we may one day support a selection foreground color like Xterm does, it may be a good idea to reserve that slot now, so we've always got our foreground and background colors paired in the same order.

Reply:
Actually, I love that. That ship may have sailed for the cursor color, however... as i have always wanted to add cursor foreground.

Since we didn't have palette report support before (unreleased) 1.22[^1], is there still time to make changes in our reserved color indices?

[^1]: under the assumption that we want them stable now that people can back up and restore them :)

Reply:
> That ship may have sailed for the cursor color, however... as i have always wanted to add cursor foreground.

Cursor color is a bit of a special case, because it can be thought of as both foreground and background depending on the style. When it's the block style, you think of the cursor color as a background attribute, and when it's the underline or bar style it's more like a foreground attribute.

But if you consider the way cursors were originally implemented, it makes the most sense as a foreground attribute, because the block cursor was really just implemented as a reverse attribute toggle. So if your default text is white on black, and your cursor color is say green, then an underline cursor is going to be green on black, and a block cursor is just green on black reversed. In both case the black background is retained, and the foreground white is replaced with green.

That said, if you wanted to introduce a second cursor color, referring to the two as foreground and background is going to be confusing no matter which way you look at it, and maybe it would be easiest to refer to the second color and something generic like "secondary cursor color".

> Since we didn't have palette report support before (unreleased) 1.22, is there still time to make changes in our reserved color indices?

I don't think we want to mess with the default text and frame indices, because even though we couldn't query them, we've already been telling people they can set the colors at those specific positions. The cursor color is probably less of an issue, because I don't think we've publicaly mentioned its value anywhere. But as I explained above, I don't think we necessarily should be changing that anywav - just reserve a slot for a "secondary cursor color" if you think we're going to need that.

Reply:
Oh, that makes way more sense. Thanks for explaining.

Reply:
Do you not still want to reserve a spot for a `SECONDARY_CURSOR_COLOR` before the new selection color entries, or have you decided we aren't likely to need that?

Reply:
Eh, I was going to leave it. But you're right, I might as well.

Reply:
Leonard's position is more (paraphrased by me) "If these aren't a committed public API, we should be able to renumber them at our leisure later." I have not fully digested how I feel about Terminal Preview and DECCTR.

Reply:
Yeah, I'm OK with that too.